### PR TITLE
Start using new cost model with Guardian credits

### PR DESCRIPTION
**Pull request by jamesgorrie**
We need to start moving to the new rights model.

Part of the problem while I am going through is that we are promoting people to use the credit/source model to set the cost of the image.

If we don;t switch over we are going to be digging a hole deeper for our selves.  

This PR switches over so that we use the `usageRights` info and the fact that we've marked an image with a Guardian creditas free.

--- 

2015: 535474 of 535812 (99.8%)
2014: 135586  of 136023 (99.6%)
2013: 102453 of 103185 (99.2%)
2012: 96689 of 100713 (96%)

### CODE REVIEW

**Comment by theefer** [comment]
Note that presumably a lot should be matched once we ingested the Picdar rights. I'm keen to do this ASAP but waiting for the reindex to finish. Hopefully can start tomorrow.

**Comment by jamesgorrie** [comment]
Ready to go :boom::boom::boom:

**Comment by jamesgorrie** [comment]
Could we have a decision on this code wise so when we get through the philosophy - we can go?

File: media-api/app/lib/elasticsearch/SearchFilters.scala [hunk 1]
```diff
@@ -16,9 +16,40 @@ trait SearchFilters extends ImageFields {
...
+  // We're showing `Conditional` here too as we're considering them potentially
+  // free. We could look into sending over the search query as a cost filter
+  // that could take a comma separated list e.g. `cost=free,conditional`.
+  val freeUsageRightsFilter = freeToUseCategories.toNel.map(filters.terms(usageRightsField("category"), _))
+
+  val freeFilter = filterOrFilter(freeSupplierFilter, freeUsageRightsFilter)
+  val nonFreeFilter = freeFilter.map(filters.not)
+
+  val guardianCreditFilter = guardianCredits.toNel.map(cs => filters.terms(metadataField("credit"), cs))
+  val freeFilterOrGuardianCredits = filterOrFilter(freeFilter, guardianCreditFilter)
```
Comment (theefer) [inline media-api/app/lib/elasticsearch/SearchFilters.scala]: Can we make it clear this is a temporary hack?

File: common-lib/src/main/scala/com/gu/mediaservice/lib/config/UsageRightsConfig.scala [hunk 2]
```diff
@@ -57,4 +57,19 @@ object UsageRightsConfig {
     "Getty Images" -> payGettySourceList
   )
 
+  // FIXME: There must be a better way (._.).
+  // Potentially making cost a lookup again?
+  val freeToUseCategories: List[String] = List(
```
Comment (theefer) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/config/UsageRightsConfig.scala]: I swear we already had this list somewhere?

**Comment by theefer** [comment]
:+1:

File: common-lib/src/main/scala/com/gu/mediaservice/lib/config/UsageRightsConfig.scala (hunk 2 above)
Comment (jamesgorrie) [inline common-lib/src/main/scala/com/gu/mediaservice/lib/config/UsageRightsConfig.scala]: It was moved here a while ago.

File: media-api/app/lib/elasticsearch/SearchFilters.scala (hunk 1 above)
Comment (jamesgorrie) [inline media-api/app/lib/elasticsearch/SearchFilters.scala]: Yip - done.

**Comment by theefer** [comment]
:+1:

**Comment by theefer** [comment]
:tada:

**Pull request merged**